tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/DataStreamsCheckpointerTests.cs

```diff
@@ -0,0 +1,103 @@
+// <copyright file="DataStreamsCheckpointerTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DataStreamsMonitoring;
+using Datadog.Trace.Headers;
+using Datadog.Trace.TestHelpers;
+using FluentAssertions;
+using Moq;
+using Xunit;
+
+namespace Datadog.Trace.Tests.DataStreamsMonitoring;
+
+public class DataStreamsCheckpointerTests
+{
+    [Fact]
+    public void WhenDisabled_ReturnsNoop()
+    {
+        // setup a tracer that has the right config
+        var tracerSettings = TracerSettings.Create(new Dictionary<string, object>() { { ConfigurationKeys.DataStreamsMonitoring.Enabled, false } });
+        var tracer = new Tracer(tracerSettings, agentWriter: null, sampler: null, scopeManager: null, statsd: null);
+        TracerRestorerAttribute.SetTracer(tracer);
+
+        var chkpt = DataStreamsCheckpointer.Get();
+
+        chkpt.Should().NotBeNull();
+        chkpt.Should().BeOfType<NoopCheckpointer>();
+    }
+
+    [Fact]
+    public void WhenEnabled_ReturnsDefault()
+    {
+        var tracerSettings = TracerSettings.Create(new Dictionary<string, object>() { { ConfigurationKeys.DataStreamsMonitoring.Enabled, true } });
+        var tracer = new Tracer(tracerSettings, agentWriter: null, sampler: null, scopeManager: null, statsd: null);
+        TracerRestorerAttribute.SetTracer(tracer);
+
+        var chkpt = DataStreamsCheckpointer.Get();
+
+        chkpt.Should().NotBeNull();
+        chkpt.Should().BeOfType<DataStreamsCheckpointer>();
+    }
+
+    [Fact]
+    public void ShouldCheckpointOnConsume()
+    {
+        var tracerSettings = TracerSettings.Create(new Dictionary<string, object>() { { ConfigurationKeys.DataStreamsMonitoring.Enabled, true } });
+        var tracer = new Tracer(tracerSettings, agentWriter: null, sampler: null, scopeManager: null, statsd: null);
+        TracerRestorerAttribute.SetTracer(tracer);
+        var chkpt = DataStreamsCheckpointer.Instance;
```

Comment:
It is being tested above already, but maybe use `Get` here as well to imitate customer behavior more closely